rules/TypeDeclaration/ValueObject/AddParamTypeForFunctionLikeWithinCallLikeArgFromArgDeclaration.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\TypeDeclaration\ValueObject;
+
+use PHPStan\Type\ObjectType;
+use Rector\Validation\RectorAssert;
+
+/**
+ * @see \Rector\Tests\TypeDeclaration\Rector\FunctionLike\AddParamTypeForFunctionLikeWithinCallLikeDeclarationRector\AddParamTypeForFunctionLikeWithinCallLikeDeclarationRectorTest
+ */
+final readonly class AddParamTypeForFunctionLikeWithinCallLikeArgFromArgDeclaration
+{
+    /**
+     * @param int<0, max>|string $callLikePosition
+     * @param int<0, max> $functionLikePosition
+     * @param int<0, max>|string $fromArgPosition
+     */
+    public function __construct(
+        private string $className,
+        private string $methodName,
+        private int|string $callLikePosition,
+        private int $functionLikePosition,
+        private int|string $fromArgPosition,
+        private bool $onlyAcceptClassString,
+    ) {
+        RectorAssert::className($className);
+    }
+
+    public function getObjectType(): ObjectType
+    {
+        return new ObjectType($this->className);
+    }
+
+    public function getMethodName(): string
+    {
+        return $this->methodName;
+    }
+
+    /**
+     * @return int<0, max>|string
+     */
+    public function getCallLikePosition(): int|string
+    {
+        return $this->callLikePosition;
+    }
+
+    /**
+     * @return int<0, max>
+     */
+    public function getFunctionLikePosition(): int
+    {
+        return $this->functionLikePosition;
+    }
+
+    /**
+     * @return int<0, max>|string
+     */
+    public function getFromArgPosition(): int|string
+    {
+        return $this->fromArgPosition;
+    }
+
+    public function onlyAcceptClassString(): bool
+    {
+        return $this->onlyAcceptClassString;
+    }
```

Comment:
How useful is this practise? I'd expect type should be filled in any case

Reply:
I've solved this by just splitting the rules into two unique ones.

The problem was without a mechanism to seperate the two, Class const would mean sometimes you might want a string for a type, it just seemed messy to not be specific